internal/service/networkcontainer/resource_network_container_test.go

```diff
@@ -36,16 +34,26 @@ func TestAccNetworkContainerRS_basicAWS(t *testing.T) {
 		CheckDestroy:             checkDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: configAWS(projectName, orgID, cidrBlock, constant.AWS, "US_EAST_1"),
+				Config: configBasic(projectID, cidrBlock, constant.AWS, "US_EAST_1"),
 				Check: resource.ComposeTestCheckFunc(
 					checkExists(resourceName),
 					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
 					resource.TestCheckResourceAttr(resourceName, "provider_name", constant.AWS),
 					resource.TestCheckResourceAttrSet(resourceName, "provisioned"),
+
+					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "provider_name", constant.AWS),
+					resource.TestCheckResourceAttrSet(dataSourceName, "provisioned"),
+
+					resource.TestCheckResourceAttrWith(dataSourcePluralName, "results.#", acc.IntGreatThan(0)),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.id"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.atlas_cidr_block"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.provider_name"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.provisioned"),
```

Comment:
is this a trend we're going to use in all resources? testing data sources within the resource test? Or is there something special about this?

Depending on the answer: could we have some comments? (always having in mind readability)

Reply:
it's not really a new pattern, we have both approaches depending on the resource. We do the same tests to the resource and datasource (checking the same attributes) and we create the same resource to test the datasource as the "resource basic test". 
i'll take it to the tech meeting

Reply:
A small detail to consider it that when we have acceptance tests that test both the resource and data source you will notice that the test name does not include TestAcc\***RS**\_\*, and instead defines TestAcc*_* ([example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/74687c3088486c3b0c7bb92c72fd03d041a6d6af/internal/service/advancedcluster/resource_advanced_cluster_test.go#L23))

Reply:
thx, good point